Signal/src/ViewControllers/ConversationView/MessagesViewController.m

```diff
@@ -2267,41 +2234,10 @@ - (void)collectionView:(JSQMessagesCollectionView *)collectionView
 
     self.page = MIN(self.page + 1, (NSUInteger)kYapDatabaseMaxPageCount - 1);
 
-    // To update a YapDatabaseViewMappings, you can call either:
-    //
-    // * [YapDatabaseViewMappings updateWithTransaction]
-    // * [YapDatabaseViewMappings getSectionChanges:rowChanges:forNotifications:withMappings:]
-    //
-    // ...but you can't call both.
-    //
-    // If ensureDynamicInteractionsForThread modifies the database,
-    // the mappings will be updated by yapDatabaseModified.
-    // This will leave the mapping range in a bad state.
-    // Therefore we temporarily disable observation of YapDatabaseModifiedNotification
-    // while updating the range and the dynamic interactions.
-    [[NSNotificationCenter defaultCenter] removeObserver:self name:YapDatabaseModifiedNotification object:nil];
-
-    // We need to update the dynamic interactions after loading earlier messages,
-    // since the unseen indicator may need to move or change.
-    [self ensureDynamicInteractions];
+    self.shouldObserveDBModifications = NO;
```

Comment:
Good point.  I've pulled out `resetMappings` methods in both VCs to make the intent more clear.